src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,139 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private static final String iOutOfBounds = "Index is out of bounds";
+    private Node<T> head;
+    private int size;
```

Comment:
Solution without tail is interesting but not efficient and violates requirement
https://github.com/mate-academy/jv-linked-list/blob/master/checklist.md#dont-iterate-your-nodes-from-the-head-to-the-tail-or-vice-versa
Please introduce tail and use it to improve performance of logic (add to end of list operation is faster when working with tail and also getting node by index will be faster if index is in 2nd part of list and we iterating from tail)